harfbuzz/src/blob.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::{mem, ops, ptr, slice};
+use std::os::raw::{c_char, c_uint};
+use sys;
+
+/// Wrapper type that owns an `hb_blob_t`.
+pub struct Blob {
+    raw: *mut sys::hb_blob_t,
+}
+
+impl Blob {
+    /// Create a new read-only blob.
+    pub fn new_read_only(data: &[u8]) -> Self {
+        assert!(data.len() <= c_uint::max_value() as usize);
+        unsafe {
+            Blob::from_raw(sys::hb_blob_create(
+                data.as_ptr() as *const c_char,
+                data.len() as c_uint,
+                sys::HB_MEMORY_MODE_READONLY,
+                ptr::null_mut(), // user data
+                None, // destroy callback
+            ))
+        }
+    }
+
+    /// Construct a `Blob` from a raw pointer. Takes ownership of the blob.
+    pub unsafe fn from_raw(raw: *mut sys::hb_blob_t) -> Self {
+        Blob { raw }
+    }
+
+    /// Returns the size of the blob in bytes.
+    pub fn len(&self) -> usize {
+        unsafe { sys::hb_blob_get_length(self.raw) as usize }
+    }
+
+    /// Make this blob immutable.
+    pub fn make_immutable(&mut self) {
+        unsafe { sys::hb_blob_make_immutable(self.raw); }
+    }
+
+    /// Returns true if the blob is immutable.
+    pub fn is_immutable(&mut self) {
```

Comment:
This looks like a copy paste error ... it is the same as the above function.

Reply:
Oops! Fixed.